olp-cpp-sdk-dataservice-write/tests/StreamLayerClientImplTest.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <matchers/NetworkUrlMatchers.h>
+#include <mocks/CacheMock.h>
+#include <mocks/NetworkMock.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/dataservice/write/generated/model/ResponseOkSingle.h>
+#include "StreamLayerClientImpl.h"
+
+namespace {
+
+using namespace testing;
+using namespace olp;
+using namespace olp::dataservice::write;
+using namespace olp::tests::common;
+
+constexpr int64_t kTwentyMib = 20971520;
+const olp::client::HRN kHRN{"hrn:here:data:::catalog"};
+constexpr auto kLayerName = "layer";
+constexpr auto kContentType = "application/json";
+
+struct ResponseOkWithTraceIdFrom {
+  template <typename Request>
+  model::ResponseOkSingle operator()(Request request) {
+    model::ResponseOkSingle response;
+    response.SetTraceID(*request.GetTraceId());
+    return response;
+  }
+};
```

Comment:
Is this struct used somewhere?

Reply:
This will  be usefull down the road, when we port more calls to sync api

Reply:
YAGNI? :)

Reply:
Removed